NPE on repo update

When trying to update or create repo via dashboard, management API throws NPE and returns 500 status:
```
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "com.amihaiemil.eoyaml.YamlMapping.value(String)" because "repo" is null
java.util.concurrent.CompletionException: java.lang.NullPointerException: Cannot invoke "com.amihaiemil.eoyaml.YamlMapping.value(String)" because "repo" is null
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$UniCompose.tryFire(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(Unknown Source)
	at java.base/java.util.concurrent.CompletableFuture$AsyncSupply.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.lang.NullPointerException: Cannot invoke "com.amihaiemil.eoyaml.YamlMapping.value(String)" because "repo" is null
	at com.artipie.management.api.ApiRepoUpdateSlice.lambda$response$2(ApiRepoUpdateSlice.java:118)
	... 4 more
```

Artipie version is `v0.20`

--- a/src/main/java/com/artipie/management/api/ApiRepoUpdateSlice.java
+++ b/src/main/java/com/artipie/management/api/ApiRepoUpdateSlice.java
@@ -153,7 +153,8 @@ public final class ApiRepoUpdateSlice implements Slice {
     }
 
     /**
-     * Obtain value from payload.
+     * Obtain value from payload, payload is a query string (not url-encoded):
+     * <code>name1=value1&name2=value2</code>.
      * @param payload Payload to parse
      * @param name Parameter name to obtain
      * @return Parameter value

--- a/src/main/java/com/artipie/management/api/ApiRepoUpdateSlice.java
+++ b/src/main/java/com/artipie/management/api/ApiRepoUpdateSlice.java
@@ -119,18 +119,22 @@ public final class ApiRepoUpdateSlice implements Slice {
                                 if (type == null || !Scalar.class.isAssignableFrom(type.getClass())) {
                                     throw new IllegalStateException("Repository type required");
                                 }
-                                final YamlMapping stor = repo.yamlMapping("storage");
-                                if (stor == null) {
+                                final YamlMapping ystor = repo.yamlMapping("storage");
+                                final String sstor = repo.string("storage");
+                                if (ystor == null && sstor == null) {
                                     throw new IllegalStateException("Repository storage is required");
+                                }
+                                YamlMappingBuilder yrepo = Yaml.createYamlMappingBuilder()
+                                    .add("type", type)
+                                    .add("permissions", repo.value("permissions"));
+                                if (ystor == null) {
+                                    yrepo = yrepo.add("storage", sstor);
+                                } else {
+                                    yrepo = yrepo.add("storage", ystor);
                                 }
                                 res = CompletableFuture.completedFuture(
                                     Yaml.createYamlMappingBuilder().add(
-                                        "repo",
-                                        Yaml.createYamlMappingBuilder()
-                                            .add("type", type)
-                                            .add("storage", stor)
-                                            .add("permissions", repo.value("permissions"))
-                                            .build()
+                                        "repo", yrepo.build()
                                     ).build()
                                 );
                             }
@@ -157,7 +161,7 @@ public final class ApiRepoUpdateSlice implements Slice {
      */
     private static String value(final String payload, final String name) {
         final int start = payload.indexOf(String.format("%s=", name)) + name.length() + 1;
-        int end = payload.indexOf(';', start);
+        int end = payload.indexOf('&', start);
         if (end == -1) {
             end = payload.length();
         }

--- a/src/test/java/com/artipie/management/api/ApiRepoUpdateSliceTest.java
+++ b/src/test/java/com/artipie/management/api/ApiRepoUpdateSliceTest.java
@@ -5,6 +5,7 @@
 package com.artipie.management.api;
 
 import com.amihaiemil.eoyaml.Yaml;
+import com.amihaiemil.eoyaml.YamlMappingBuilder;
 import com.artipie.asto.Content;
 import com.artipie.asto.Key;
 import com.artipie.asto.Storage;
@@ -21,8 +22,8 @@ import java.io.IOException;
 import org.hamcrest.MatcherAssert;
 import org.hamcrest.core.IsEqual;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
 import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvSource;
 import org.junit.jupiter.params.provider.ValueSource;
 
 /**
@@ -53,8 +54,9 @@ final class ApiRepoUpdateSliceTest {
         this.storage = new InMemoryStorage();
     }
 
-    @Test
-    void createsRepoConfiguration() throws IOException {
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void createsRepoConfiguration(final boolean defstor) {
         MatcherAssert.assertThat(
             "Returns FOUND",
             new ApiRepoUpdateSlice(new FakeConfigFile(this.storage)),
@@ -68,7 +70,7 @@ final class ApiRepoUpdateSliceTest {
                         "/dashboard/%s", ApiRepoUpdateSliceTest.userRepo()
                     )
                 ),
-                new Content.From(body(ApiRepoUpdateSliceTest.REPO, "maven"))
+                new Content.From(body(ApiRepoUpdateSliceTest.REPO, "maven", defstor))
             )
         );
         MatcherAssert.assertThat(
@@ -81,13 +83,19 @@ final class ApiRepoUpdateSliceTest {
     }
 
     @ParameterizedTest
-    @ValueSource(strings = {".yaml", ".yml"})
-    void updatesRepoConfiguration(final String extension) throws IOException {
+    @CsvSource({
+        ".yaml,true",
+        ".yaml,false",
+        ".yml,true",
+        ".yml,false"
+    })
+    void updatesRepoConfiguration(final String extension, final boolean defstor)
+        throws IOException {
         final String oldtype = "pypi";
         final String type = "docker";
         this.storage.save(
             new Key.From(String.format("%s%s", ApiRepoUpdateSliceTest.userRepo(), extension)),
-            new Content.From(yaml(oldtype).getBytes())
+            new Content.From(yaml(oldtype, defstor).getBytes())
         ).join();
         MatcherAssert.assertThat(
             "Returns FOUND",
@@ -103,7 +111,7 @@ final class ApiRepoUpdateSliceTest {
                         "/dashboard/%s", ApiRepoUpdateSliceTest.userRepo()
                 )
                 ),
-                new Content.From(body(ApiRepoUpdateSliceTest.REPO, type))
+                new Content.From(body(ApiRepoUpdateSliceTest.REPO, type, defstor))
             )
         );
         MatcherAssert.assertThat(
@@ -122,31 +130,34 @@ final class ApiRepoUpdateSliceTest {
         );
     }
 
-    private static byte[] body(final String reponame, final String type) {
-        return String.format("repo=%s;config=%s", reponame, yaml(type)).getBytes();
+    private static byte[] body(final String reponame, final String type, final boolean defstor) {
+        return String.format("repo=%s&config=%s", reponame, yaml(type, defstor)).getBytes();
     }
 
     private static String userRepo() {
         return String.format("%s/%s", ApiRepoUpdateSliceTest.USER, ApiRepoUpdateSliceTest.REPO);
     }
 
-    private static String yaml(final String type) {
-        return Yaml.createYamlMappingBuilder().add(
-            "repo", Yaml.createYamlMappingBuilder()
-                .add("type", type)
-                .add(
-                    "storage",
-                    Yaml.createYamlMappingBuilder().add("type", "fs").add("path", "my/path").build()
-                )
-                .add(
-                    "permissions", Yaml.createYamlMappingBuilder().add(
-                        "john", Yaml.createYamlSequenceBuilder()
-                            .add("read")
-                            .add("write")
-                            .build()
-                    ).build()
+    private static String yaml(final String type, final boolean defstor) {
+        YamlMappingBuilder repo = Yaml.createYamlMappingBuilder()
+            .add("type", type)
+            .add(
+                "permissions", Yaml.createYamlMappingBuilder().add(
+                    "john", Yaml.createYamlSequenceBuilder()
+                        .add("read")
+                        .add("write")
+                        .build()
                 ).build()
-        ).build().toString();
+            );
+        if (defstor) {
+            repo = repo.add("storage", "default");
+        } else {
+            repo = repo.add(
+                "storage",
+                Yaml.createYamlMappingBuilder().add("type", "fs").add("path", "my/path").build()
+            );
+        }
+        return Yaml.createYamlMappingBuilder().add("repo", repo.build()).build().toString();
     }
 
 }

